sbt-dump-event-schema/src/main/scala/org/virtuslab/ash/DumpEventSchemaKeys.scala

```diff
@@ -8,7 +8,9 @@ import sbt.taskKey
 trait DumpEventSchemaKeys {
   lazy val dumpEventSchema = taskKey[Unit]("Dump event schema")
   lazy val dumpEventSchemaOutputFile = taskKey[File]("Output file to dump event schema to")
+  lazy val dumpEventSchemaOutputFilePath = settingKey[String]("Path to event schema dump file")
   lazy val dumpEventSchemaOutputFilename = settingKey[String]("Filename to dump event schema to")
+  lazy val dumpEventSchemaOutputDirectoryPath = settingKey[String]("Directory path to event schema dump")
```

Comment:
Are all 4 keys still necessary? Esp. the one `dumpEventSchemaOutputFile` that isn't likely (?) to be ever directly set by the users?

Reply:
I.e. a more general question, does **sbt's project model use keys** also for the **internal properties** of the plugin? Or are the keys considered to be more or less the "interface" of a plugin?

Reply:
Yes but no.
[This plugin](https://github.com/sbt/sbt-assembly/blob/9a2ca181226a8b437a784792343a203a0b94a6b8/src/main/scala/sbtassembly/AssemblyPlugin.scala#L76) is written by sbt creator. Interesting options are: `assemblyOutputPath` (actually a file) and `assemblyJarName`, which is used in creating `assemblyOutputPath`